.github/workflows/build.yml

```diff
@@ -0,0 +1,31 @@
+name: build
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ ubuntu-latest, windows-latest ]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install Rust nightly
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
+        override: true
+        profile: minimal
+    - name: Build
+      run: |
+        cargo build --manifest-path lib/Cargo.toml --verbose --target-dir=target
+        cargo build --manifest-path Cargo.toml --verbose --target-dir=target
```

Comment:
I didn't know `--workspace`. Will update.

The `target-dir` makes the cache shared for multiple crates (ex. do not use `lib/target`). It can speed things up a bit.

Reply:
I removed `--target-dir` as workspace seems sufficient for reusing `target/`.